### ISSUE TITLE
some confusion of iters setting.

### ISSUE THREAD

**Issue by CuberrChen** [charlesCXK/TorchSemiSeg#48]
Hi.According to the Getting Started.md, there is different epochs for different label ratio. but i'm confused about the setting's meaning.
if i'm right, here you only need to specify the total iters.It is not necessary to calculate the epoch corresponding to different label ratio?

If I want to train supervised baseline, does it mean that if my total iters are 40k and the label ratio is 0.125, then I need to do 40k iterations(maybe batch size is 8) on this 0.125*10852 labeled images only? But if i use semi supervised method, one iteration means fetching the labeled data once and fetching the unlabeled data once?

Hope to receive your reply. very thanks.

best,

**Comment by CuberrChen** [comment]
Is there a training code here for supervised learning with different label rates?
In fact, my query is that for the same iters, a smaller label rate means more iterations of this label data. For example, for Pascal voc+Aug, 20000 iters in a 1/8 label+batch size 8 configuration, those label data will be traversed 121 times. But with a full label, it is only traversed 14 times, even though more different data is seen at this point

**Comment by leewming** [comment]
i also have this issue

**Comment by charlesCXK** [comment]
@CuberrChen @leewming 
Sorry for the late reply!
As mentioned in [this document](https://github.com/charlesCXK/TorchSemiSeg/blob/main/docs/getting_started.md), we control the total iterations during training in each experiment similar (almost the same), including the supervised baseline and semi-supervised methods. Since the code is trained based on the `epoch` mode, we convert the number of iterations to number of epochs.

**Comment by charlesCXK** [comment]
I will take VOC as an example.
1. We totally have 10582 images. The full supervised baseline is trained for 60 epochs with batch size 16, thus having 10582*60/16 = 39682.5 iters.
2. If we train CPS under the 1/8 split, we have 1322 labeled images and 9260 unlabeled images. Since the number of unlabeled images is larger than the number of labeled images, the `epoch` is defined as passing all the unlabeled images to the network. In each iteration, we have 8 labeled images and 8 unlabeled images, thus having 9260/8 = 1157.5 iters in one epoch. Then the total epochs we need is 39682.5/1157.5 = 34.28 ≈ 34. 
3. For the supervised baseline under the 1/8 split, the batch size 8 (as illustrated in Appendix A in the paper) and the iteration number is 39682.5 (the same as semi-supervised method).

**Comment by CuberrChen** [comment]
thanks.It's a great job.
